src/collective/collectionfilter/baseviews.py

```diff
@@ -118,6 +120,7 @@ def input_type(self):
         else:
             return 'checkbox'
 
+    @memoize  # because we need to call this for is_available below
```

Comment:
might be huge in memory?

Reply:
It's only during the request and its only the size of the number of elements in the filter, not the whole result set. Calling it result everywhere is perhaps confusing.

Reply:
@petschki I am using the right api right? It should only last until the view is gone, ie end of request

Reply:
sorry for not responding earlier ... I do not know enough of the memoize internals how and when this gets invalidated ...